### DISCUSSION TITLE
AfterSpec() of Report is not getting executed

### DISCUSSION THREAD

**Discussion by mihirsasmal** [pactumjs/pactum#224] [Q&A] [answered]
AfterSpec() of Report is not getting executed when I am using typescript and when the test fails.

It works perfectly when used in javascript.
Here is the code in javascript :

```
const awesomeReporter = {    
    async afterSpec(spec) {
        await addMsg({message:JSON.stringify({Request: spec.request, Response: spec.response}, null, 2)});
     }
  }; 
  beforeAll(function () {
    request.setBaseUrl(BASE_URL);
    reporter.add(awesomeReporter);
  }); 
```

However, same when i convert in typescript, it does not work
Here is my code in typescript :

```
const awesomeReporter = {
    
    name :'ReporterTest',
    async afterSpec(data:SpecData) {
        
        await addMsg({message:JSON.stringify({Request: data.request, Response: data.response}, null, 2), context :'test'});
     },
     // eslint-disable-next-line @typescript-eslint/no-empty-function
     async afterStep() { },
     async afterTest() { },
     async afterInteraction() { },
     async end() { }
  };
```

Am I doing something wrong here ?
Any suggestion, how to make it work ?

**Comment by ASaiAnudeep** [comment, marked as answer]
I'm not able to reproduce the issue. The below code is working perfectly fine.

```ts
import { reporter, spec } from "pactum";

reporter.add({
  name: 's',
  afterSpec(data) {
    console.log(data.request);
  },
  afterStep(data) {
    console.log(data);
  },
  afterInteraction(data) {
    console.log(data);
  },
  afterTest(data) {
    console.log(data);
  },
  end() {

  }
});

spec()
  .get('https://github.com/pactumjs/pactum/discussions/224')
  .expectStatus(200)
  .toss()
```

**Reply by mihirsasmal** [reply]
Thanks AsaiAnudeep for your reply. 
 As you mentioned here, it is working perfectly fine. 
 I was trying to keep it in global place to use for all test files and I was not doing that correctly. 
It would be nice if any example doc or code given to use certain configuration globally for all test files like baseurl , headers, queryparams.
